var/spack/repos/builtin/packages/rocblas/0005-omp-for-test.patch

```diff
@@ -0,0 +1,38 @@
+diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
+index 679edd1..9a4a902 100644
+--- a/clients/gtest/CMakeLists.txt
++++ b/clients/gtest/CMakeLists.txt
+@@ -141,12 +141,13 @@ target_include_directories( rocblas-test
+     $<BUILD_INTERFACE:${BLAS_INCLUDE_DIR}>
+     $<BUILD_INTERFACE:${BLIS_INCLUDE_DIR}> # may be blank if not used
+     $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
++    $<BUILD_INTERFACE:${ROCM_OPENMP_EXTRAS_DIR}/include>
+ )
+ 
+ if( BUILD_FORTRAN_CLIENTS )
+-  target_link_libraries( rocblas-test PRIVATE rocblas_fortran_client )
++  target_link_libraries( rocblas-test PRIVATE "${ROCM_OPENMP_EXTRAS_DIR}/lib/libomp.so" rocblas_fortran_client )
+ endif( )
+-target_link_libraries( rocblas-test PRIVATE ${BLAS_LIBRARY} ${GTEST_BOTH_LIBRARIES} roc::rocblas )
++target_link_libraries( rocblas-test PRIVATE ${BLAS_LIBRARY} ${GTEST_BOTH_LIBRARIES} "${ROCM_OPENMP_EXTRAS_DIR}/lib/libomp.so" roc::rocblas )
+ 
+ if( CUDA_FOUND )
+   target_include_directories( rocblas-test
+@@ -159,7 +160,7 @@ if( CUDA_FOUND )
+ else( )
+   # auto set in hip_common.h
+   #target_compile_definitions( rocblas-test PRIVATE __HIP_PLATFORM_HCC__ )
+-  target_link_libraries( rocblas-test PRIVATE hip::host hip::device )
++  target_link_libraries( rocblas-test PRIVATE "${ROCM_OPENMP_EXTRAS_DIR}/lib/libomp.so" hip::host hip::device )
+ endif( )
+ 
+ if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+@@ -179,7 +180,7 @@ if (NOT WIN32)
+ else()
+   list( APPEND COMMON_LINK_LIBS "libomp")
+ endif()
+-target_link_libraries( rocblas-test PRIVATE ${COMMON_LINK_LIBS} )
++target_link_libraries( rocblas-test PRIVATE "${ROCM_OPENMP_EXTRAS_DIR}/lib/libomp.so" ${COMMON_LINK_LIBS} )
```

Comment:
Isn't `OpenMP::OpenMP_CXX` already in `${COMMON_LINK_LIBS}`? [rocBLAS adds it after doing `find_package(OpenMP)`](https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/rocm-5.4.3/clients/CMakeLists.txt#L76-L84). I definitely appreciate that the workaround that you've found, but I'm still deeply confused by how OpenMP is supposed to be used.

Reply:
@cgmb, Created a patch similar to [0001-Guard-use-of-OpenMP-to-make-it-optional.patch](https://gist.github.com/cgmb/971a24d2c8df70aefc6d35efd6f7c582).  to make openmp optional for the test.
Thanks